### PR TITLE
Nexus redundant slicing

### PR DESCRIPTION
**Pull request by dwpaley**
This was essential for processing or viewing JF16M data.

The behavior of `_dataset_as_flex` is changed if it's passed an array with first dimension 1. (that axis is no longer squeezed out). This is necessary because `get_raw_data` now pre-slices the full array, so the previous first axis passed to `_dataset_as_flex` no longer exists. This is the only place that `_dataset_as_flex` is called. Tests still pass.

### CODE REVIEW

File: src/dxtbx/nexus/__init__.py [hunk 1]
```diff
@@ -479,9 +479,8 @@ def get_raw_data(
     else:
         data = list(nxdata.values())[0]
     all_data = []
+    sliced_outer = data[index]
     for module_slices in get_detector_module_slices(nxdetector):
```
Comment (graeme-winter) [inline src/dxtbx/nexus/__init__.py:483]: I am left with the question as to why it was ever written like this 🤔

File: newsfragments/565.bugfix [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Improved speed of reading many-panel Nexus images by optimizing array slicing
```
Comment (graeme-winter) [inline newsfragments/565.bugfix:1]: Suspect @ndevenish may want to tinker with the newsfragment?

**Review by graeme-winter — APPROVED** [review]
Looks good I think? It feels a bit odd that it was originally implemented that way - I think a 2 line comment above the `for module_slices in...` line explaining what this is actually doing would be appreciated by future archeologists spelunking in `dxtbx`

**Comment by graeme-winter** [comment]
@dwpaley would be really useful to have an indication of the time difference as a result of merging this PR, for permanent record and general interest.

**Comment by dwpaley** [comment]
Tested by reading an image from Nexus master files pointing to 3000 images on the Bernina JF16M:
```
$ cat > ~/time_read_data.py
from dxtbx.model.experiment_list import ExperimentListFactory
import sys
import time
el = ExperimentListFactory.from_filenames([sys.argv[1]], load_models=False)
t0 = time.time()
el[0].imageset.get_raw_data(0)
print(time.time()-t0)

dwpaley@login07:/global/cfs/cdirs/m3562
$ for n in {1..6}; do libtbx.python ~/time_read_data.py sf_bernina_data_p20231/assembled/110/acq000$n.h5; done
9.87814736366272
9.986922025680542
9.886291265487671
9.849736213684082
9.74486756324768
9.779048919677734

[...cherry-pick commit 72857b8b7...]

dwpaley@login07:/global/cfs/cdirs/m3562
$ for n in {1..6}; do libtbx.python ~/time_read_data.py sf_bernina_data_p20231/assembled/110/acq000$n.h5; done
0.18897414207458496
0.2043781280517578
0.20243358612060547
0.1900634765625
0.19011616706848145
0.1889021396636963
```

**Comment by dwpaley** [comment]
Note this is co-authored-by @irisdyoung. If you all merge this on your end could you please add:
Co-authored-by: Iris Young <irisdyoung@gmail.com>
Thanks!

**Comment by graeme-winter** [comment]
> Tested by reading an image from Nexus master files pointing to 3000 images on the Bernina JF16M:
> 
> ```
> $ cat > ~/time_read_data.py
> from dxtbx.model.experiment_list import ExperimentListFactory
> import sys
> import time
> el = ExperimentListFactory.from_filenames([sys.argv[1]], load_models=False)
> t0 = time.time()
> el[0].imageset.get_raw_data(0)
> print(time.time()-t0)
> 
> dwpaley@login07:/global/cfs/cdirs/m3562
> $ for n in {1..6}; do libtbx.python ~/time_read_data.py sf_bernina_data_p20231/assembled/110/acq000$n.h5; done
> 9.87814736366272
> 9.986922025680542
> 9.886291265487671
> 9.849736213684082
> 9.74486756324768
> 9.779048919677734
> 
> [...cherry-pick commit 72857b8b7...]
> 
> dwpaley@login07:/global/cfs/cdirs/m3562
> $ for n in {1..6}; do libtbx.python ~/time_read_data.py sf_bernina_data_p20231/assembled/110/acq000$n.h5; done
> 0.18897414207458496
> 0.2043781280517578
> 0.20243358612060547
> 0.1900634765625
> 0.19011616706848145
> 0.1889021396636963
> ```

That's quite the speed up, thank you

**Comment by dwpaley** [comment]
I was glad it worked so well :)
For the newsfragment do you happen to mean removing the "how"? I.e. just write "improved speed of reading many-panel Nexus images"?

**Pull request merged**